src/Mod/Path/PathScripts/PathOp.py

```diff
@@ -99,246 +103,254 @@ class ObjectOp(object):
-
-        # members being set later
+        # class instance attributes being set later
         self.commandlist = None
         self.horizFeed = None
         self.horizRapid = None
-        self.job = None
         self.model = None
         self.radius = None
         self.stock = None
         self.tool = None
         self.vertFeed = None
         self.vertRapid = None
-        self.addNewProps = None
+        # instantiate class attributes separately that are
+        #   necessary here and upon document restore
+        self._initAttributes(obj)
+
+        # Add properties to current operation object
+        self.initProperties(obj)
 
+        # Operation-specific sub-class continuation of this __init__() method
         self.initOperation(obj)
 
+        # Set default values unless otherwise flagged
+        # The default values must be ready prior to this section.
         if not hasattr(obj, 'DoNotSetDefaultValues') or not obj.DoNotSetDefaultValues:
-            job = self.setDefaultValues(obj)
+            job = PathUtils.addToJob(obj)
             if job:
+                self.setDefaultValues(obj, job, init=True)
                 job.SetupSheet.Proxy.setOperationProperties(obj, name)
                 obj.recompute()
                 obj.Proxy = self
 
-    def setEditorModes(self, obj, features):
-        '''Editor modes are not preserved during document store/restore, set editor modes for all properties'''
+        self.setEditorModes(obj)
 
-        for op in ['OpStartDepth', 'OpFinalDepth', 'OpToolDiameter', 'CycleTime']:
-            if hasattr(obj, op):
-                obj.setEditorMode(op, 1)  # read-only
+    def _initAttributes(self, obj):
+        """_initAttributes(obj) ...
+        Sets attributes necessary for new class instance,
+        as well as on document restore action."""
+        self.job = None
+        self.addNewProps = None
+        self.editorModes = list()
+        self.propertiesReady = False
+        self.propDefaults = None
+        self.features = self.opFeatures(obj)
+        self.tmpDebugGrp = None
+        self.showDebugObjects = False
+        self._initOpAttributes(obj)
+
+    def initProperties(self, obj, warn=False):
+        '''initProperties(obj) ... Create operation properties for
+        the current operation object based on a list of tuples - each
+        formated as (prop type, name, group, tooltip).'''
+        self.addNewProps = list()
+
+        propDefs = self.propertyDefinitions()
+        for (prtyp, nm, grp, tt) in propDefs:
+            if not hasattr(obj, nm):
+                obj.addProperty(prtyp, nm, grp, tt)
+                self.addNewProps.append(nm)
+
+        # Set enumeration lists for enumeration properties
+        newPropCnt = len(self.addNewProps)
+        if newPropCnt > 0:
+            ENUMS = self.propertyEnumerations()
```

Comment:
I don't think the use of all-caps variable names is appropriate here (also several other places).

Reply:
Like Markus says, all caps is usually reserved for constants; it is set at the beginning of the file, and then it is not modified any more. If you are assigning this variable through a function, then it's not really constant.